### PR TITLE
docs(hooks): add `Highlight` component

### PR DESCRIPTION
**Pull request by francoischalifour**
This adds the `<Highlight>` component to our Hooks example.

It supports regular attributes, nested attributes, and array attributes.

## Usage

```js
function Hit({ hit }) {
  return (
    <>
      {/* regular attribute */}
      <Highlight hit={hit} attribute="name" />
      {/* nested attribute */}
      <Highlight
        hit={hit}
        attribute={['hierarchicalCategories', 'lvl0']}
        className="Hit-label"
      />
    </>
  );
}

export function App(props) {
  return (
    <InstantSearch {...props}>
      <Hits hitComponent={Hit} />
    </InstantSearch>
  );
}
```

### CODE REVIEW

File: yarn.lock [hunk 1]
```diff
@@ -12567,6 +12567,22 @@ instantsearch.css@7.3.1:
   resolved "https://registry.yarnpkg.com/instantsearch.css/-/instantsearch.css-7.3.1.tgz#7ab74a8f355091ae040947a9cf5438f379026622"
   integrity sha512-/kaMDna5D+Q9mImNBHEhb9HgHATDOFKYii7N1Iwvrj+lmD9gBJLqVhUw67gftq2O0QI330pFza+CRscIwB1wQQ==
 
+instantsearch.js@4.33.2:
```
Comment (Haroenv) [inline yarn.lock:12570]: the dependency in React InstantSearch hooks should be `^` so this doesn't get duplicated

File: examples/hooks/components/Highlight.tsx [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+  getPropertyByPath,
+} from 'instantsearch.js/es/lib/utils';
+import React from 'react';
+
+import { cx } from '../cx';
+
+type HighlightPartProps = {
+  children: React.ReactNode;
+  highlightedTagName: string;
+  nonHighlightedTagName: string;
```
Comment (Haroenv) [inline examples/hooks/components/Highlight.tsx:13]:
```suggestion
```suggestion
  highlightedTagName: React.ReactType;
  nonHighlightedTagName: React.ReactType;
```
```

File: examples/hooks/components/Highlight.tsx [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+
+function HighlightPart({
+  children,
+  highlightedTagName,
+  isHighlighted,
+  nonHighlightedTagName,
+}: HighlightPartProps) {
+  const TagName = isHighlighted
+    ? (highlightedTagName as 'span')
+    : (nonHighlightedTagName as 'span');
```
Comment (Haroenv) [inline examples/hooks/components/Highlight.tsx:25]:
```suggestion
  const TagName = isHighlighted ? highlightedTagName : nonHighlightedTagName;
```

File: examples/hooks/components/Highlight.tsx [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+      {children}
+    </TagName>
+  );
+}
+
+export type HighlightProps<THit> = {
+  hit: THit;
+  attribute: keyof THit | string[];
+  highlightedTagName?: string;
+  nonHighlightedTagName?: string;
```
Comment (Haroenv) [inline examples/hooks/components/Highlight.tsx:44]:
```suggestion
  highlightedTagName?: React.ReactType;
  nonHighlightedTagName?: React.ReactType;
```

**Review by Haroenv — COMMENTED** [review]
implementation lgtm

File: examples/hooks/components/Highlight.tsx (hunk 2 above)
Comment (francoischalifour) [reply examples/hooks/components/Highlight.tsx]: Thank you!! Was wondering if that type existed.

See 55f73bff7e078841577046ca321f5227cbe60514.

File: examples/hooks/components/Highlight.tsx (hunk 3 above)
Comment (francoischalifour) [reply examples/hooks/components/Highlight.tsx]: See 55f73bff7e078841577046ca321f5227cbe60514.

File: examples/hooks/components/Highlight.tsx (hunk 4 above)
Comment (francoischalifour) [reply examples/hooks/components/Highlight.tsx]: See 55f73bff7e078841577046ca321f5227cbe60514.

File: yarn.lock (hunk 1 above)
Comment (francoischalifour) [reply yarn.lock]: I deduped the lock file in d6bac79b7761c70b98670941fcb0f6add9b6ba82.

**Review by Haroenv — APPROVED** [review]

**Pull request merged**